DUI3-DX/Sdk/Speckle.Connectors.Utils/Builders/TraversalExtensions.cs

```diff
@@ -0,0 +1,26 @@
+﻿using Speckle.Core.Models;
+using Speckle.Core.Models.GraphTraversal;
+
+namespace Speckle.Connectors.Utils.Builders;
+
+public static class TraversalExtensions
+{
+  public static IEnumerable<TraversalContext> TraverseWithProgress(
+    this GraphTraversal traversalFunction,
+    Base rootObject,
+    Action<string, double?>? onOperationProgressed,
+    CancellationToken cancellationToken = default
+  )
+  {
+    var traversalGraph = traversalFunction.Traverse(rootObject).ToArray();
+    int count = 0;
+    foreach (var tc in traversalGraph)
+    {
+      cancellationToken.ThrowIfCancellationRequested();
+
+      yield return tc;
+
+      onOperationProgressed?.Invoke("Converting", (double)++count / traversalGraph.Length);
```

Comment:
Should we really report to UI progress when traversing? If so I believe it shouldn't be named with "Converting"?

Reply:
See other comment, this isn't traversal progress (because traversal is super quick)
This is the conversion progress (as the `IEnumerable` is being enumerated)

Let me know if you want me to explain this better over voice call.